nomenclature/validation.py

```diff
@@ -19,6 +19,9 @@ def is_subset(x, y):
 def validate(nc, df):
     """Validation of an IamDataFrame against codelists of a Nomenclature"""
 
+    if not isinstance(df, IamDataFrame):
+        df = IamDataFrame(df)
```

Comment:
At this conversion attempt we assume that converting into an IamDataFrame is possible. Should we do checks before trying to convert or does the constructor of the IamDataFrame take care of that?

Reply:
I think it is helpful for the user to directly see which kind of error caused the error when `df` cannot be cast to an IamDataFrame